app/Libraries/UsernameValidation.php

```diff
@@ -95,7 +96,10 @@ public static function validateUsersOfUsername(string $username): ValidationErro
             }
 
             // top 100
-            if ($user->rankHighests->contains(fn ($r) => $r->rank <= 100)) {
+            // Queried directly on model because User::rankHighests is disabled
+            // when experimental rank is set as default.
+            $highestRank = RankHighest::where('user_id', $user->getKey())->min('rank');
```

Comment:
Does this work outside the loop with `whereIn(..., $users` if it's using `min` since as long any previous user meets the requirement it should be locked?

Reply:
that seems applicable to the others as well but I'm changing those here 🤔